helm/athena/templates/ingress.yaml

```diff
@@ -13,6 +13,10 @@ metadata:
     cert-manager.io/cluster-issuer: {{ .Values.ingress.tls.clusterIssuer }}
     {{- if .Values.security.subnet.restrictAccess.gsAPI }}
     nginx.ingress.kubernetes.io/whitelist-source-range: {{ template "whitelistCIDR" . }}
```

Comment:
This PR is primarily interested in the `nginx.ingress.kubernetes.io/whitelist-source-range` annotation, that is why we only enable it when that annotation isn't already added by that deprecated (REST API/GSAPI, vintage-only, rarely used) mechanism.

Apparently there's currently only one deployment still making use of `.Values.security.subnet.restrictAccess.gsAPI = true` and once that goes away, one might remove that functionality from the chart.